hardware_interface/include/hardware_interface/helpers/component_lists_management.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__HELPERS__COMPONENT_LISTS_MANAGEMENT_HPP_
+#define HARDWARE_INTERFACE__HELPERS__COMPONENT_LISTS_MANAGEMENT_HPP_
+
+#include <string>
+#include <vector>
+
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+namespace helpers
+{
+
+/**
+  * \brief Get values for queried_interfaces from the int_values. int_values data structure matches
+  * int_interfaces vector.
+  *
+  * \param values values to return.
+  * \param queried_interfaces interfaces for which values are queried.
+  * \param int_interfaces full list of interfaces of a component.
+  * \param int_values internal values of a component.
+  * \return return_type::INTERFACE_VALUE_SIZE_NOT_EQUAL if values and queried_interfaces arguments
+  * do not have the same length; return_type::INTERFACE_NOT_FOUND if one of queried_interfaces is
+  * not defined in int_interfaces; return return_type::INTERFACE_NOT_PROVIDED if queried_interfaces
+  * list is is empty; return_type::OK otherwise.
+  */
+return_type get_internal_values(
```

Comment:
shouldn't they all be `inline`?

Reply:
YES! That is what I was searching for :) Thanks

(It was too late yesterday...)